README.md

```diff
@@ -8,23 +8,34 @@
 
-*Micro XRCE-DDS* implements a client-server protocol to enable resource-constrained devices (clients) to take part in DDS communications.
-*Micro XRCE-DDS Agent* (server) makes possible this communication.
-The *Micro XRCE-DDS Agent* acts on behalf of the *Micro XRCE-DDS Clients* and enables them to take part as DDS publishers and/or subscribers in the DDS Global Data Space.
-*Micro XRCE-DDS* provides both, a plug and play *Micro XRCE-DDS Agent* and an API layer which allows you to implement your *Micro XRCE-DDS Clients*.
+<p align="center"> <img src="docs/Concept.png" alt="Image" width="90%"/> </p>
 
-<p align="center"> <img src="docs/xrcedds_architecture.png" alt="Image"/> </p>
+The *Micro XRCE-DDS Clients* request operations to the *Agent* to publish and/or subscribe to topics in the DDS global dataspace. Remote procedure calls, as defined by the [DDS-RPC standard](https://www.omg.org/spec/DDS-RPC/About-DDS-RPC/), are also supported, allowing *Clients* to communicate in the DDS dataspace according to a request/reply paradigm.
+The *Agents* process these requests and send back a response with the operation status result and with the requested data, in the case of subscribe/reply operations.
 
-This repository contains the totality of *Micro XRCE-DDS* products:
+*eProsima Micro XRCE-DDS* provides the user with a C API to create *Micro XRCE-DDS Clients* applications. The library can be configured at compile-time via a set of CMake flags allowing to enable or disable some profiles before compilation, and to manipulate several parameters controlling some of the library's functionalities, which in turn allow tuning the library size.
 
-<p align="center"> <img src="docs/general_architecture.png" alt="Image"/> </p>
+The *Micro XRCE-DDS Agent* receives messages containing request operations from the *Clients*, processes these requests and sends back a response with the operation status result and with the requested data, in the case of subscribe/reply operations.
+
+*Agents* keep track of the *Clients* by means of a dedicated `ProxyClient` entity that acts on behalf of the latter.
+This is made possible by the creation of *DDS Entities* on the *Agent* as a result of *Clients*' operations, such as *Participants*, *Topics*, *Publishers*, and *Subscribers*, which can interact with the DDS global dataspace.
+
+The communication between a *Micro XRCE-DDS Client* and a *Micro XRCE-DDS Agent* is achieved by means of several kinds of built-in transports: **UDPv4**, **UDPv6**, **TCPv4**, **TCPv6** and **Serial** communication. In addition, there is the possibility for the user to generate its own **Custom** transport.
+
+<p align="center"> <img src="docs/General.png" alt="Image" width="70%"/> </p>
+
+This repository contains the totality of the *eProsima Micro XRCE-DDS* products:
+
+- *Micro XRCE-DDS Client*
+- *Micro XRCE-DDS Client*
+- *Micro XRCE-DDS Gen*
```

Comment:
I personally would expect this list ot be clickable. Therefore, write something as:
* [Micro XRCE-DDS Client](https://github.com/eProsima/Micro-XRCE-DDS-Client)